File: python/leetcode/125. Valid Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if str == "":
            return True
        start = 0
        end = len(s) - 1

        while start < end:
            if s[start].isalnum() and s[end].isalnum():
                if s[start].casefold() == s[end].casefold():
                    start += 1
                    end -= 1
                else:
                    return False

            if not s[start].isalnum():
                start += 1
            if not s[end].isalnum():
                end -= 1

        return True

File: c/leetcode/125. Valid Palindrome.c


bool isPalindrome(char *s)
{
  if (strlen(s) == 0)
    return true;
  int start = 0;
  int end = strlen(s);

  while (start < end)
  {
    if (isalnum(s[start]) && isalnum(s[end]))
    {
      if (tolower(s[start]) == tolower(s[end]))
      {
        start++;
        end--;
      }
      else
        return false;
    }

    if (!isalnum(s[start]))
      start++;
    if (!isalnum(s[end]))
      end--;
  }

  return true;
}
